File: src/api/sys/job.rs
use axum::routing::{delete, get, post, put};
use crate::api::web_path::{WebPath, WebPathType};
use crate::application::sys::*;

pub fn sys_job() -> WebPath {
    WebPath::new()
        .route(
            "/list",
            WebPathType::Get,
            Some("获取定时任务列表"),
            get(s_sys_job::list),
        )
        .route(
            "/add",
            WebPathType::Post,
            Some("添加定时任务"),
            post(s_sys_job::add),
        )
        .route(
            "/edit",
            WebPathType::Put,
            Some("编辑定时任务"),
            put(s_sys_job::edit),
        )
        .route(
            "/del",
            WebPathType::Delete,
            Some("删除定时任务"),
            delete(s_sys_job::delete),
        )
        .route(
            "/validate_cron",
            WebPathType::Post,
            Some("验证cron表达式"),
            post(s_sys_job::validate_cron),
        )
        .route(
            "/hand_execute_job",
            WebPathType::Post,
            Some("执行定时任务"),
            post(s_sys_job::hand_execute_job),
        )
}

pub fn sys_job_info() -> WebPath {
    WebPath::new()
        .route(
            "/list",
            WebPathType::Get,
            Some("获取定时任务日志列表"),
            get(s_sys_job_log::list),
        )
        .route(
            "/add",
            WebPathType::Post,
            Some("添加定时任务日志"),
            post(s_sys_job_log::add),
        )
}

File: frontend/src/api/modules/monitor/job.ts
import axios from "@/api";

// ==================== 定时任务 API ====================

// 获取定时任务列表
export const getJobListAPI = (params?: {
  job_name?: string;
  job_group?: string;
  status?: string;
  page_num?: number;
  page_size?: number;
}) => {
  return axios({
    url: "/sys/job/list",
    method: "get",
    params
  });
};

// 添加定时任务
export const addJobAPI = (data: {
  job_name: string;
  job_group: string;
  task_type: string;
  task_count: number;
  run_count: number;
  cron_expression: string;
  job_params?: string;
  status: string;
  remark: string;
}) => {
  return axios({
    url: "/sys/job/add",
    method: "post",
    data
  });
};

// 编辑定时任务
export const editJobAPI = (data: {
  job_id: string;
  job_name: string;
  job_group: string;
  task_type: string;
  task_count: number;
  run_count: number;
  cron_expression: string;
  job_params?: string;
  status: string;
  remark: string;
}) => {
  return axios({
    url: "/sys/job/edit",
    method: "put",
    data
  });
};

// 删除定时任务
export const deleteJobAPI = (params: { job_id: string }) => {
  return axios({
    url: "/sys/job/del",
    method: "delete",
    params
  });
};

// 验证cron表达式
export const validateCronAPI = (data: { cron_expression: string }) => {
  return axios({
    url: "/sys/job/validate_cron",
    method: "post",
    data
  });
};

// 手动执行定时任务
export const handExecuteJobAPI = (data: { job_id: string }) => {
  return axios({
    url: "/sys/job/hand_execute_job",
    method: "post",
    data
  });
};

// ==================== 定时任务日志 API ====================

// 获取定时任务日志列表
export const getJobLogListAPI = (params?: {
  job_id?: string;
  page_num?: number;
  page_size?: number;
}) => {
  return axios({
    url: "/sys/jobinfo/list",
    method: "get",
    params
  });
};
